src/main/java/org/mondemand/Client.java

```diff
@@ -1207,11 +1211,14 @@ private synchronized void dispatchContextStats()
         }
       }
     }
-    contextStats.clear();
   }
 
   public ConcurrentHashMap<ContextList, AtomicLongMap<String>> getContextStats() {
     return contextStats;
   }
 
+  public ConcurrentHashMap<String, SamplesMessage> getSamples() {
```

Comment:
I would not add a method just to be used in the unit test. use reflection in the unit test to access samples.